combine/combine.go

```diff
@@ -0,0 +1,171 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package combine
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/cluster"
+	"github.com/obolnetwork/charon/eth2util/keystore"
+	tblsv2 "github.com/obolnetwork/charon/tbls/v2"
+)
+
+// Combine combines validator keys contained in inputDir, and writes the original BLS12-381 private keys.
+// Combine is validator-aware: it'll recombine all the validator keys listed in the "Validator" field of the lock file.
+// To do so, the user must prepare inputDir as follows:
+//   - place the lock file in input dir, named as "cluster-lock.json"
+//   - create one directory for each operator, named after their ENR
+//   - place in each of those directories the content of the "validator_keys" directory, contained in their Charon runtime
+//     directory
+func Combine(ctx context.Context, inputDir string, force bool) error {
+	lfPath := filepath.Join(inputDir, "cluster-lock.json")
+	b, err := os.Open(lfPath)
+	if err != nil {
+		return errors.Wrap(err, "read lock file")
+	}
+
+	var lock cluster.Lock
+	if err := json.NewDecoder(b).Decode(&lock); err != nil {
+		return errors.Wrap(err, "unmarshal lock file")
+	}
```

Comment:
you can do what we do here: https://github.com/ObolNetwork/charon/blob/bf0f4694d0e4a505105d2175138b010146464252/app/disk.go#L29

Reply:
Implemented!